src/mime.cr

```diff
@@ -5,31 +5,77 @@ require "crystal/system/mime"
 # ```
 # require "mime"
 #
-# MIME.from_extension(".html")         # => "text/html"
-# MIME.from_filename("path/file.html") # => "text/html"
+# MIME.from_extension(".html")         # => "text/html; charset=utf-8"
+# MIME.from_filename("path/file.html") # => "text/html; charset=utf-8"
 # ```
 #
 # The registry is populated with some default values (see `DEFAULT_TYPES`)
 # as well as the operating system's MIME database.
 #
+# ## OS-provided MIME database
+#
+# On a POSIX system, the following files are tried to be read in sequential order,
+# stopping at the first existing file. These values override those from `DEFAULT_TYPES`.
+#
+# ```plain
+# /etc/mime.types
+# /etc/httpd/mime.types                    # Mac OS X
+# /etc/httpd/conf/mime.types               # Apache
+# /etc/apache/mime.types                   # Apache 1
+# /etc/apache2/mime.types                  # Apache 2
+# /usr/local/etc/httpd/conf/mime.types
+# /usr/local/lib/netscape/mime.types
+# /usr/local/etc/httpd/conf/mime.types     # Apache 1.2
+# /usr/local/etc/mime.types                # FreeBSD
+# /usr/share/misc/mime.types               # OpenBSD
+# ```
+#
+# ## Registering custom MIME types
+#
 # Applications can register their own MIME types:
 #
 # ```
+# require "mime"
 # MIME.from_extension?(".cr")     # => nil
 # MIME.extensions("text/crystal") # => []
```

Comment:
This needs to be updated with reference to the returned `Set` instead of `Array`.